src/bootstrap/config.rs

```diff
@@ -1349,3 +1374,116 @@ fn threads_from_config(v: u32) -> u32 {
         n => n,
     }
 }
+
+/// Returns the commit to download, or `None` if we shouldn't download CI artifacts.
+fn download_ci_rustc_commit(download_rustc: Option<StringOrBool>, verbose: bool) -> Option<String> {
+    // If `download-rustc` is not set, default to rebuilding.
+    let if_unchanged = match download_rustc {
+        None | Some(StringOrBool::Bool(false)) => return None,
+        Some(StringOrBool::Bool(true)) => false,
+        Some(StringOrBool::String(s)) if s == "if-unchanged" => true,
+        Some(StringOrBool::String(other)) => {
+            panic!("unrecognized option for download-rustc: {}", other)
+        }
+    };
+
+    // Handle running from a directory other than the top level
+    let top_level = output(Command::new("git").args(&["rev-parse", "--show-toplevel"]));
+    let top_level = top_level.trim_end();
+    let compiler = format!("{top_level}/compiler/");
+    let library = format!("{top_level}/library/");
+
+    // Look for a version to compare to based on the current commit.
+    // Only commits merged by bors will have CI artifacts.
+    let merge_base = output(Command::new("git").args(&[
+        "rev-list",
+        "--author=bors@rust-lang.org",
+        "-n1",
+        "--first-parent",
+        "HEAD",
+    ]));
+    let commit = merge_base.trim_end();
+    if commit.is_empty() {
+        println!("error: could not find commit hash for downloading rustc");
```

Comment:
Should these be `eprintln!()` instead?

Reply:
Bootstrap is unfortunately quite inconsistent about which it uses :( I don't think it's worth worrying about currently, but I can open an issue to settle on one of the other.
```
$ rg -g '!build.rs' println src/bootstrap/ | grep -v eprintln | wc -l
88
$ rg eprintln src/bootstrap/ | wc -l
60
```